tools/screenshot-test/src/app/result/result.component.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+import {FirebaseService} from '../firebase.service';
+
+@Component({
+  selector: 'app-result',
+  templateUrl: './result.component.html',
+  styleUrls: ['./result.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ResultComponent {
+
+  /** Test name, display on top of the result card */
+  testName: string;
+
+  /** Test, diff and golden image urls */
+  testImageUrl: string;
+  diffImageUrl: string;
+  goldImageUrl: string;
+
+  /** Test result, auto set collapse to be the same as the result value */
+  @Input()
+  get result() {
+    return this._result;
+  }
+  set result(value: boolean) {
+    this._result = value;
+    this.collapse = value;
+  }
+  _result: boolean;
+
+  get resultText() {
+    return this._result ? 'Passed' : 'Failed';
+  }
+
+  /** Collapse: whether collapse or expand the card to show images */
+  @Input() collapse: boolean = true;
+
+  /** Mode: the result card has three modes, flip, side by side, and diff */
+  @Input()
+  get mode() {
+    return this._mode;
+  }
+  set mode(value: 'flip' | 'side' | 'diff') {
+    this._mode = value;
+    this.modeEvent.emit(value);
+    this._changeDetectorRef.markForCheck();
+  }
+  _mode: 'flip' | 'side' | 'diff' = 'diff';
+
+  /** When mode is "flip" whether we show the test image or the golden image */
+  @Input()
+  get flipping() {
+    return this._flipping;
+  }
+
+  set flipping(value: boolean) {
+    this._flipping = value;
+    this.flippingEvent.emit(value);
+    this._changeDetectorRef.markForCheck();
+  }
+   _flipping: boolean = false;
+
+  get collapseIcon() {
+    return this.collapse ? 'keyboard_arrow_right' : 'keyboard_arrow_down';
+  }
+
+   /** Filename, the test file name. */
+  @Input()
+  get filename() {
+    return this._filename;
+  }
+  set filename(filename: string) {
```

Comment:
Removed the getter since we don't need it. Added a function `setFilenameAndFetchImages`.